nncf/onnx/graph/nncf_graph_builder.py

```diff
@@ -21,11 +21,198 @@
+    Second, add weight port ids determined dynamically if metatype could have them.
+
+    :param node: ONNX node.
+    :param onnx_graph: ONNXGraph.
+    :return: Port ids with weights.
+    """
+    port_ids = set()
+    metatype = get_metatype(onnx_graph, node)
+    constant_port_ids = get_constant_weight_port_ids(metatype)
+    port_ids.update(constant_port_ids)
+    possible_port_ids = get_possible_weight_port_ids(metatype)
+    for port_id in possible_port_ids:
+        if _get_weight_edge_name(onnx_graph, node, port_id):
+            port_ids.add(port_id)
+    return port_ids
+
+
+def _is_node_with_bias(node: onnx.NodeProto, onnx_graph: ONNXGraph) -> bool:
+    """
+    Returns True if node has bias tensor, otherwise - False.
+
+    :param node: ONNX node.
+    :param onnx_graph: ONNXGraph.
+    :return: True if node has bias tensor, otherwise - False.
+    """
+    metatype = get_metatype(onnx_graph, node)
+    if metatype in OPERATIONS_WITH_BIAS_METATYPES:
+        bias_tensor_port_id = get_bias_tensor_port_id(metatype)
+        if len(node.input) > bias_tensor_port_id:
+            return True
+    return False
+
+
+def _get_weight_attr(node: onnx.NodeProto, onnx_graph: ONNXGraph, weight_port_id: int) -> Dict[str, str]:
+    """
+    Returns weight attributes.
+
+    :param node: ONNX node.
+    :param onnx_graph: ONNXGraph.
+    :param weight_port_ids: Port ids with weights location.
+    :return: Weight attributes.
+    """
+    weight_attrs = {}
+    weight_edge_name = _get_weight_edge_name(onnx_graph, node, weight_port_id)
+    edge = onnx_graph.get_edge(weight_edge_name)
+    weight_shape = ONNXGraph.get_edge_shape(edge)
+    weight_attrs[weight_port_id] = {"name": weight_edge_name, "shape": weight_shape}
+    return weight_attrs
+
+
+def _get_gemm_attrs(node: onnx.NodeProto) -> Dict[str, int]:
+    """
+    Returns transpose attrbiutes of GEMM node.
+
+    :param node: GEMM node.
+    :return: Trnaspose attributes.
+    """
+    gemm_attrs = {"transA": 0, "transB": 0}
+    attribute_names = ["transA", "transB"]
+    for attr in node.attribute:
+        if attr.name in attribute_names:
+            gemm_attrs[attr.name] = onnx.helper.get_attribute_value(attr)
+    return gemm_attrs
+
+
+def _get_node_attrs(node: onnx.NodeProto, onnx_graph: ONNXGraph) -> Dict[str, Any]:
+    """
+    Returns node attributes.
+
+    :param node: Node.
+    :param onnx_graph: ONNXGraph.
+    :return : Node attributes.
+    """
+    metatype = get_metatype(onnx_graph, node)
+    if metatype == ONNXGemmMetatype:
+        return _get_gemm_attrs(node)
+
+
+def _get_bias_attr(node: onnx.NodeProto, onnx_graph: ONNXGraph) -> Dict[str, str]:
+    """
+    Returns bias tensor attributes.
+
+    :param node: ONNX node.
+    :param onnx_graph: ONNXGraph.
+    :return: Bias tensor attributes.
+    """
+    bias_attrs = {}
+    metatype = get_metatype(onnx_graph, node)
+    if _is_node_with_bias(node, onnx_graph):
```

Comment:
Why don't we have this check outside the method? I mean, for each node, even without bias, it would contain some `bias_attrs`.

Reply:
And this is the same for `node_attrs`. The logic is that if it is an empty `Dict` then there is no such attribute.